src/main/scala/com/codacy/client/bitbucket/client/BitbucketClientBase.scala

```diff
@@ -264,13 +262,6 @@ abstract class BitbucketClientBase(credentials: Credentials) {
     }
   }
 
-  private def withClient[T](block: WSClient => T): Try[T] = {
-    val client: WSClient = buildClient()
-    val result = Try(block(client))
-    client.close()
```

Comment:
Who is now responsible for closing the client? It seems that I can create a `BitBucketClient` using default parameters and that inner client will never be closed and I don't have access to it to close it myself.

Should we care that it's not closed?

Reply:
Just made the client a val for it to be closeable from the outside since @andreaTP would require more work to make it optional.

Reply:
Can you add documentation about closing it? Maybe something like: "This class will not close `client` even when it instantiates it. Callers of this class are responsible for closing it."

It's not that important, so I'll approve it anyway.